java/ql/lib/semmle/code/java/regex/regex.qll

```diff
@@ -510,9 +513,15 @@ abstract class RegexString extends StringLiteral {
    * ```
    */
   private predicate flag(string c) {
```

Comment:
Probably out of scope, but this predicate might not behave as desired when the flags of a group don't affect the whole pattern, e.g. for `a(?i:b)` it would have the flag `i` as result even though this only affects the `b`.